### ISSUE TITLE
Jakarta Package Rename support will require a fork of spring

### ISSUE THREAD

**Issue by hazendaz** [psi-probe/psi-probe#1674]
In order to implement a compliant version that works for tomcat 10, spring would need forked and changed to jakarta.  The Spring team has stated they have no plans to do so any time soon.  Spring 6 isn't even slated to have that support.  They are waiting on all larger implementation releases but reality for this project is we need this way sooner as we will lose support for tomcat 10 if we do nothing.

I suspect I'll make a fork of spring for this purpose.  Personally I don't find that all that difficult and not sure why spring isn't doing so as it's mostly just a package rename with some really old cleanup that probably doesn't even affect spring to start with.  At the moment there seem to be a lot of users interested in this but little interest from spring.  I do understand their point as the bulk of the world isn't going to flip over night and we are years probably from that.  However, this library is entirely built on spring and as such I'm going to have to do this work or look at dropping spring or just lose users.  None of this particularly excites me here just note to self.

The work otherwise for tomcat10 is already done locally on my machine for this project.  I started that some months back.  I'm just stuck due to spring and based on comments they have in past week or two, its clear what I have to do.  That said, I'll be putting the fork if it pans out to be easy up to the psi probe namespace rather than my personal and if others find us of it great.  If not, not a huge concern but I can control alignment easier that way and make it clear.

Assuming that isn't difficult as well, at the moment I have this all mixed together as one repo.  I'll likely start to split this into pieces or worst case I'm just entirely duplicating packages to make it easy.

**Comment by pizzi80** [comment]
have you considered trying tomcat 10 migration tool?

> Applications that run on Tomcat 9 and earlier will not run on Tomcat 10 without changes. Java EE based applications designed for Tomcat 9 and earlier may be placed in the $CATALINA_BASE/webapps-javaee directory and Tomcat will automatically convert them to Jakarta EE and copy them to the webapps directory. This conversion is performed using the Apache Tomcat migration tool for Jakarta EE tool which is also available as a separate download for off-line use.

**Comment by hazendaz** [comment]
This product is spring based, so spring needs to support it.

**Comment by pizzi80** [comment]
> the Apache Tomcat migration tool for Jakarta EE tool which is also available as a separate download for off-line use.

You could try to download the Apache tomcat migration tool and run it over spring's source code ...

> Assuming that isn't difficult as well, at the moment I have this all mixed together as one repo. I'll likely start to split this into pieces or worst case I'm just entirely duplicating packages to make it easy.

I think that it would be better to clone the actual "psi-probe" into "psi-probe-legacy" and migrate the code of the actual version to jakarta ee which will be compatible only for tomcat 10+

Once people will port their app to jakarta ee they will never go back so there will be no need to maintain older version

my 2 cents

**Comment by hazendaz** [comment]
That would not help.  That would require that I support spring code too.  I tried a year back to migrate spring myself since they also don't wan't to support 2 versions (still was same stance back in October).  Its too extensive, the tool would not get it all, and ultimately it would be not worth the effort.  Tomcat doesn't even expect anyone to use tomcat 10 in any production mode so the point is sort of mute right now.  Basically I'm going to wait until spring supports it here.  They are aware of the porting tools as well provided by EE.

**Comment by hazendaz** [comment]
When the proper support comes along the idea would be to just outright replace after some period of dual support.  So I'd likely just use same repo here rather than split it up.  I've done the same with other repos that work on EE that I support and was able to release support of as the couping with spring didn't exist.

**Comment by pizzi80** [comment]
You could evaluate to start to migrate from Spring to JSF 3 with low priority and then upgrade the JSF's jars to the upcoming JSF 4 wich will be a major refactor and cleanup of JSF. 

Considering that:
1) JSP is basically dead
2) JSF 4 will be the de facto java web mvc framework supported by the Eclipse foundation
3) For a project like psi-probe you will need few dependencies and a minimal configuration
4) You can use Servlets 
5) You have basic and form authentication
6) Runs on vanilla tomcat
7) Fast development of web views

Take a look at:
[https://github.com/eclipse-ee4j/mojarra](https://github.com/eclipse-ee4j/mojarra)

Probably it would be nice to compare the actual psi-probe memory footprint versus a minimal JSF4 web app to check if it's a viable solution for psi-probe to migrate

**Comment by hazendaz** [comment]
@pizzi80 Since spring has offiically stated they are picking up all jakarta for spring 6, the fact that spring boot works with cdi via joinfaces, I'm not sure there is much value here tryign to progress faster than next year.  Eclipse is supporting both currently and making patches as needed still so I think we are ok here.  I'd love though to see something done here with the very legacy javascript in this project.  Updating that to jquery would be a huge help.  I'm not very good on the front end and couldn't ever figure it out.  I'd also like to see JSP / JSTL go away. Being more of a java geek myself, I'd rather just use JSF / EL for that.  Then toss in joinfaces so spring works and its all good with both EE and Spring.  But all of these come with considerable amount of work and I'm not using this enough myself any more to justify doing the work.  But I'm certainly open to someone else picking it up :)

**Comment by hazendaz** [comment]
closing this issue as spring 6 addresses and we will just migrate once that is out.  fork no longer being explored as a result.

**Comment by pizzi80** [comment]
https://spring.io/blog/2022/11/16/spring-framework-6-0-goes-ga

**Comment by hazendaz** [comment]
Any ongoing tracking for dependencies are done [here](https://github.com/psi-probe/psi-probe/issues/2205).  This will not happen fast.  Spring getting there was great but pieces still coming out with patches in recent days.  Additionally, the entire stack of any support has to be looked at as jakarta means death nail to older support due to how ingrained javax support is.  Additionally, we have to drop jdk 8 and 11 support to do this.
